File: src/zork/models/talk/Dialog.java
package zork.models.talk;

import java.util.List;

/**
 * Prototype Dialog class for use with dialog System. WIP.
 *
 * @author d.peters
 * @version 0.2
 */
public class Dialog {
    /**
     * List of dialog choices.
     */
    private List<Message> choiches;

    /**
     * Owner of this dialog. The owner object must implement ITalkAble
     * interface.
     */
    private ITalkAble owner;
}

File: src/main/kotlin/ch/d/peters/zork/models/talk/Dialog.kt
package ch.d.peters.zork.models.talk

import ch.d.peters.zork.interfaces.ITalkAble
import ch.d.peters.zork.models.talk.Message

/**
 * Prototype Dialog class for execute with dialog System. WIP.
 *
 * @author Daniel Peters
 * @version 1.1.0
 */
class Dialog(val choiches: List<Message>, val owner: ITalkAble)
